From: Steven Whitehouse <swhiteho@redhat.com>
Date: Mon, 15 Dec 2008 17:25:46 +0000
Subject: [gfs2] add UUID to gfs2 super block
Message-id: 1229361946.9571.412.camel@quoit
O-Subject: [RHEL 5.4] bz 242696: Add UUID to gfs2 super block
Bugzilla: 242696

This is a patch from upstream to add a UUID to the GFS2 superblock. Its
very small since this is all the kernel support thats required, the rest is
in userspace. All this does is reserve some space in the super block for it.

Bugzilla: #242696

diff --git a/include/linux/gfs2_ondisk.h b/include/linux/gfs2_ondisk.h
index c3c19f9..14d0df0 100644
--- a/include/linux/gfs2_ondisk.h
+++ b/include/linux/gfs2_ondisk.h
@@ -118,7 +118,11 @@ struct gfs2_sb {
 
 	char sb_lockproto[GFS2_LOCKNAME_LEN];
 	char sb_locktable[GFS2_LOCKNAME_LEN];
-	/* In gfs1, quota and license dinodes followed */
+
+	struct gfs2_inum __pad3; /* Was quota inode in gfs1 */
+	struct gfs2_inum __pad4; /* Was licence inode in gfs1 */
+#define GFS2_HAS_UUID 1
+	__u8 sb_uuid[16]; /* The UUID, maybe 0 for backwards compat */
 };
 
 /*
